### PR TITLE
feat(message transformation): implement dryrun endpoint

### PR DESCRIPTION
**Pull request by thalesmg**
Follow up to https://github.com/emqx/emqx/pull/13199

Fixes https://emqx.atlassian.net/browse/EMQX-12517

Release version: e5.7.1

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_message_transformation/src/emqx_message_transformation_http_api.erl [hunk 1]
```diff
@@ -143,6 +149,25 @@ schema("/message_transformations/reorder") ->
                 }
         }
     };
+schema("/message_transformations/test") ->
```
Comment (zmstone) [inline apps/emqx_message_transformation/src/emqx_message_transformation_http_api.erl:152]: ?
```suggestion
schema("/message_transformations/dryrun") ->
```

Comment (thalesmg) [reply apps/emqx_message_transformation/src/emqx_message_transformation_http_api.erl]: :checkered_flag:

File: apps/emqx_message_transformation/src/emqx_message_transformation_http_api.erl [hunk 2]
```diff
@@ -267,6 +292,29 @@ fields(reorder) ->
...
+    [
+        {client_attrs, mk(map(), #{required => true})},
+        {payload, mk(binary(), #{required => true})},
+        {qos, mk(range(0, 2), #{required => true})},
+        {retain, mk(boolean(), #{required => true})},
+        {topic, mk(binary(), #{required => true})},
+        {user_property,
+            mk(
+                typerefl:alias("map(binary(), binary())", user_property()),
+                #{required => true}
```
Comment (zmstone) [inline apps/emqx_message_transformation/src/emqx_message_transformation_http_api.erl:315]: why `required => true` for all ?

Comment (thalesmg) [reply apps/emqx_message_transformation/src/emqx_message_transformation_http_api.erl:315]: It's better to be explicit about the input, I think.

Comment (zmstone) [reply apps/emqx_message_transformation/src/emqx_message_transformation_http_api.erl:315]: Is the user forced to add all fields in order to trigger a dryrun ?

Comment (thalesmg) [reply apps/emqx_message_transformation/src/emqx_message_transformation_http_api.erl:315]: Yes.

Comment (thalesmg) [reply apps/emqx_message_transformation/src/emqx_message_transformation_http_api.erl:315]: Most of them don't have a good default, I think,

**Review by zmstone — APPROVED** [review]

**Review by zmstone — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2633)

**Pull request merged**